lib/src/widgets/search_bar.dart

```diff
@@ -0,0 +1,113 @@
+                  TextEditingController fieldTextEditingController,
+                  FocusNode fieldFocusNode,
+                  VoidCallback onFieldSubmitted) {
+                return TextField(
+                  decoration: InputDecoration(
+                      border: InputBorder.none,
+                      hintText: StringData.hintTextSearch),
+                  controller: fieldTextEditingController,
+                  focusNode: fieldFocusNode,
+                  style: const TextStyle(
+                      fontWeight: FontWeight.bold, color: Colors.black),
+                );
+              },
+              optionsBuilder: (TextEditingValue textEditingValue) {
+                if (textEditingValue.text.isEmpty) {
+                  return const Iterable<Coins>.empty();
+                }
+                return widget.coinslist.where((Coins option) {
+                  return option.name
+                          .toString()
+                          .toLowerCase()
+                          .contains(textEditingValue.text.toLowerCase()) ||
+                      option.symbol
+                          .toString()
+                          .toLowerCase()
+                          .contains(textEditingValue.text.toLowerCase());
+                });
+              },
+              optionsViewBuilder: (context, onSelected, options) {
+                return Align(
+                  alignment: Alignment.topLeft,
+                  child: Container(
+                    color: Colors.white,
+                    width: widthOfBoxSeletionSearchBar,
+                    child: ListView.builder(
+                      itemCount: options.length,
+                      itemBuilder: (BuildContext context, int index) {
+                        final Coins option = options.elementAt(index);
+                        return GestureDetector(
+                          onTap: () {
+                            onSelected(option);
+                          },
+                          child: Card(
+                            child: ListTile(
+                              leading: Image.network(
+                                option.image,
+                                errorBuilder: (context, error, strackTrace) =>
+                                    Icon(Icons.error),
+                              ),
+                              title: Text(
+                                option.name,
+                                style: Theme.of(context).textTheme.headline6,
+                              ),
+                            ),
+                          ),
+                        );
+                      },
+                    ),
+                  ),
+                );
+              },
+              onSelected: (Coins coin) {
+                Navigator.push(
+                    context,
+                    MaterialPageRoute(
+                        builder: (context) => DetailScreen(coin)));
```

Comment:
We defined routes with named constants, we should use them.

Reply:
fixed